static/js/message_edit.js

```diff
@@ -112,12 +109,27 @@ export function get_editability(message, edit_limit_seconds_buffer = 0) {
         //       other message updates.  This commit changed the result
         //       from FULL to NO, since the prior implementation was
         //       buggy.
-        return editability_types.NO;
+        return false;
     }
 
     // Locally echoed messages are not editable, since the message hasn't
     // finished being sent yet.
     if (message.locally_echoed) {
+        return false;
+    }
+
+    if (currently_echoing_messages.has(message.id)) {
```

Comment:
Maybe worth adding a comment "Messages where we're currently locally echoing an edit not yet acknowledged by the server".